scalyr_agent/line_matcher.py

```diff
@@ -125,6 +169,135 @@ def _readline( self, file_like, max_length=0 ):
         partial = len( line) > 0 and line[-1] != '\n' and line[-1] != '\r'
         return line, partial
 
+class JsonLineMatcher( LineMatcher ):
+    """
+    Parses a line as JSON
+
+    If the `parse_as_json` configuration item is True then
+    """
+    def __init__( self, max_line_length=5*1024, line_completion_wait_time=5*60, read_page_size=60*1024, json_message_field='log', json_timestamp_field="time" ):
+        LineMatcher.__init__( self, max_line_length, line_completion_wait_time )
+
+        self._json_message_field = json_message_field
+        self._json_timestamp_field= json_timestamp_field
+
+    def _truncated_log_line( self, line, max_length ):
+        """
+        Truncates a line if it is greater than `max_length`, and appends a message saying the number of bytes truncted
```

Comment:
I don't think we should be adding the truncate message into the actual log content here.  It's questionable, but it could seem like we are impacting the integrity of their data.  In the past, I think we just silently truncate.. or am I wrong about that?